safe-global/web-core
# feat: Safe Apps Sign and Execute transactions

## What it solves

Resolves https://github.com/safe-global/web-core/issues/454

## How this PR fixes it
We are adding the required components to allow Safe Apps to send transactions and sign messages
  - Transaction Modal components
  - Sign Message Modal components
  - Communicator support for `sendTransactions`, `signMessage` and `signTypedMessage`
  - Adding support for Safe Apps request Id's in the current EventBus. Required for closing requests from the SDK when proposing transactions
  - Adding support for components as Stepper titles (label)

## How to test it
1) When trying to interact with a contract doing an unique operation it should show the SafeTxModal with the details of the operation (Single Call)

![image](https://user-images.githubusercontent.com/1576776/189165718-82fe3e50-a630-48a3-81e3-d1deabf4516d.png)

2) When trying to interact with a contract with several operations it should show the SafeTxModal with the details of the operation actions (and create a MultiSend delegate call)

![image](https://user-images.githubusercontent.com/1576776/189166465-6a5c00ad-302c-4413-a078-5486797bdc86.png)

3) When trying to sign a message `signMessage` it should show the SignMessage Modal with the text message and the `signMessage` method

![image](https://user-images.githubusercontent.com/1576776/189166636-fd16d95c-9267-440e-ac4a-6c47d2da2724.png)


4) When trying to sign a typed message `signTypedMessage` it should show the SignMessage Modal with the text message and the `signTypedMessage`method

![image](https://user-images.githubusercontent.com/1576776/189166828-b9e75102-8468-46fa-b5b4-0f8e1c425e20.png)

## Analytics changes
